app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,55 @@
-# app/controllers/posts_controller.rb
 class PostsController < ApplicationController
-  # GET /posts or /posts.json
   def index
-    # Code to fetch and display a list of posts by a given user
+    @posts = Post.includes(:author).where(author_id: params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
   end
 
-  # GET /posts/1 or /posts/1.json
   def show
-    # Code to fetch and display details of a single post
+    @post = Post.includes(:author).find_by(author_id: params[:user_id], id: params[:id])
+
+    if @post
+      @user = @post.author
+      @comments = @post.comments
+    else
+      flash[:alert] = 'Post not found'
+      redirect_to user_posts_path(params[:user_id])
+    end
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      puts @user
+      puts @post.errors.full_messages
+      flash.now[:errors] = 'Invalid post!'
+      render :new
+    end
+  end
```

Comment:
- [ ] I was unable to create a new post on behalf of the `current_user` successfully. Please kindly address it. A video is attached to this comment to give more clarity on how to resolve it. 🙏 


https://github.com/firstpj/BLOG-appllication/assets/103386183/5cd936b8-5cb9-4f85-ae28-eeb439aa31e7